src/main/java/com/gmail/nossr50/events/skills/taming/McMMOPlayerTameEntityEvent.java

```diff
@@ -0,0 +1,67 @@
+package com.gmail.nossr50.events.skills.taming;
+
+import com.gmail.nossr50.datatypes.experience.XPGainReason;
+import com.gmail.nossr50.datatypes.skills.PrimarySkillType;
+import com.gmail.nossr50.events.experience.McMMOPlayerExperienceEvent;
+import com.google.common.base.Preconditions;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Called when a player has tamed an entity, and we are about to award them experience for it.
+ */
+public class McMMOPlayerTameEntityEvent extends McMMOPlayerExperienceEvent {
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+
+    private float xpGained;
+    private final Entity tamedEntity;
+
+    @ApiStatus.Internal
+    public McMMOPlayerTameEntityEvent(@NotNull Player player, float xp, @NotNull Entity tamedEntity) {
```

Comment:
I would expect the `McMMOPlayer` to be dependency injected into this event, with a getter (no setter).